0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1201 @@
+
+| Name                   | Type                 | Description                  |
+|:-----------------------|:---------------------|:-----------------------------|
+| `client_id`            | [UUID][]             | Client-provided UUID for this resource. MUST be unique within the ledger. |
+| `ledger`               | [URL][]              | Resource identifier for the ledger where the transfer occurs. MUST be an HTTP(S) URL where you can [get the ledger metadata][Get Ledger Metadata]. |
+| `debit_account`        | [URL][]              | Resource identifier for the account debited in this transfer. MUST be an HTTP(S) URL where you can [get the account resource][Get Account]. |
+| `credit_account`       | [URL][]              | Resource identifier for the account credited in this transfer. MUST be an HTTP(S) URL where you can [get the account resource][Get Account]. This can be the same as the `debit_account`. (Transfers where the `credit_account` is the `debit_account` are used for ILP transfers to sub-ledgers.) |
+| `amount`               | String               | The [amount][] of the currency/asset being transferred. This is debited from the `debit_account` and credited to the `credit_account` when the transfer executes. |
+| `execution_condition`  | [Crypto-Condition][] | _(Optional)_ The condition for executing the transfer. If omitted, the transfer executes unconditionally. |
+| `expires_at`           | [Date-Time][]        | _(Optional)_ The date when the transfer expires and can no longer be executed. |
+| `additional_info`      | Object               | _(Optional)_ Arbitrary fields attached to this transfer. (For example, the IDs of related transfers in other systems.) |
+| `memo`                 | Object               | _(Optional)_ Arbitrary data provided by the `debit_account` for the `credit_account`. The connector stores the ILP Header in this field. |
+| `note_to_self`         | Object               | _(Optional)_ Arbitrary data provided by the `debit_account` for itself. This field MUST be hidden when not authenticated as the `debit_account` or an admin. |
+| `id`                   | [URL][]              | _(Server-provided)_ Primary resource identifier for this transfer. MUST be an HTTP(S) URL where you can [get the transfer resource][Get Transfer]. |
+| `fulfillment`          | [URL][]              | (Server-provided) Path to the fulfillment for this transfer. MUST be an HTTP(S) URL where the client can [submit the fulfillment][Submit Fulfillment] or [get the fulfillment][Get Transfer Fulfillment]. MUST be provided if and only if this transfer has an `execution_condition`. |
+| `transfer_rejection`   | [URL][]              | (Server-provided) Path where a client can [reject this transfer][Reject Transfer]. MUST be an HTTP(S) URL. |
```

Comment:
Wouldn't  rejection_webhook  be more 'standard' (https://en.wikipedia.org/wiki/Webhook) ?

I don't have a global picture about how transfers can be rejected using such URL. Is there any RFCs explaining it?

 As I understand "cancelling" and "rejecting" are two different terms in ILP:

- "cancelling" means creating a cancellation fulfilment matching a cancellation crypto-condition. 

- rejecting means that at some point the execution fulfilment didn't match the execution condition or there was a timeout. 

DOUBT:  How this 'transfer_rejection' match in the ILP protocol. Does it means the (sender originator?) ledger must fulfil the cancellation fulfilment ? In such case transfer_cancellation (or cancellation_webhook) will be more appropiate.
(Actually I didn't find anything related in the RFCs, only the code in the five-bells implementation)